### PR TITLE
[CI] Fetch secrets from vault.

### PR DESCRIPTION
**Pull request by higuoxing**
This patch teaches concourse to fetch secrets from vault.

### CODE REVIEW

File: concourse/pipeline/res_def.yml [hunk 1]
```diff
@@ -21,23 +21,23 @@ resources:
 - name: diskquota_pr
   type: pull-request
   # We should rely on the webhook. See README if webhook doesn't work
-  webhook_token: ((diskquota-webhook-token))
+  webhook_token: ((extension/diskquota-webhook-token))
```
Comment (beeender) [inline concourse/pipeline/res_def.yml:24]: How does concourse know where is the vault server?

File: concourse/fly.sh [hunk 2]
```diff
@@ -108,9 +108,7 @@ set -v
     sp \
     -p "${pipeline_name}" \
     -c "${yml_path}" \
-    -l "${workspace}/gp-continuous-integration/secrets/gpdb_common-ci-secrets.yml" \
-    -l "${workspace}/gp-continuous-integration/secrets/gp-extensions-common.yml" \
-    -l "${workspace}/gp-continuous-integration/secrets/gpdb_6X_STABLE-ci-secrets.prod.yml" \
+    -l "${workspace}/gpdb6/concourse/vars/common_prod.yml" \
```
Comment (beeender) [inline concourse/fly.sh:111]: hmmm, what is needed from this file? `gpdb6` doesn't feel common for most of us

File: concourse/fly.sh [hunk 3]
```diff
@@ -108,9 +108,7 @@ set -v
     sp \
     -p "${pipeline_name}" \
     -c "${yml_path}" \
-    -l "${workspace}/gp-continuous-integration/secrets/gpdb_common-ci-secrets.yml" \
```
Comment (fanfuxiaoran) [inline concourse/fly.sh:111]: The values of the vars in the pipeline are saved in gpdb repo?

Comment (higuoxing) [reply concourse/fly.sh:111]: No, it's a private repo.

Comment (fanfuxiaoran) [reply concourse/fly.sh:111]:
```
 -l "${workspace}/gpdb6/concourse/vars/common_prod.yml"
```
It looks like in the gpdb6 repo, the name should be changed, a little confusing.

File: concourse/fly.sh (hunk 2 above)
Comment (higuoxing) [reply concourse/fly.sh:111]: `gcs-bucket` is needed in our pipeline.

File: concourse/fly.sh (hunk 3 above)
Comment (higuoxing) [reply concourse/fly.sh:111]: Yes, there are some commonly used variables stored in that file and it's public. There's no secret in that file.

Comment (beeender) [reply concourse/fly.sh:111]: > -l https://github.com/greenplum-db/gpdb/blob/master/concourse/vars/common_prod.yml

This actually looks nice! Does this work?

Comment (beeender) [reply concourse/fly.sh:111]: `-l` doesn't support http :(

**Review by beeender — APPROVED** [review]
:clap:

**Comment by higuoxing** [comment]
CI is updated.

**Pull request merged**